Library/Homebrew/cask/lib/hbc/cli/upgrade.rb

```diff
@@ -0,0 +1,82 @@
+module Hbc
+  class CLI
+    class Upgrade < AbstractCommand
+      option "--greedy", :greedy, false
+      option "--quiet",  :quiet, false
+      option "--force", :force, false
+      option "--force-update", :force_update, false
+      option "--skip-cask-deps", :skip_cask_deps, false
+
+      def initialize(*)
+        super
+        self.verbose = ($stdout.tty? || verbose?) && !quiet?
+      end
+
+      def run
+        outdated_casks = casks(alternative: -> { Hbc.installed }).select { |cask| cask.outdated?(greedy?) }
+
+        return if outdated_casks.empty?
+
+        oh1 "Upgrading #{Formatter.pluralize(outdated_casks.length, "outdated package")}, with result:"
+        puts outdated_casks.map { |f| "#{f.full_name} #{f.version}" } * ", "
+
+        outdated_casks.each do |old_cask|
+          odebug "Started upgrade process for Cask #{old_cask}"
+          raise CaskNotInstalledError, old_cask unless old_cask.installed? || force?
+
+          unless old_cask.installed_caskfile.nil?
+            # use the same cask file that was used for installation, if possible
+            old_cask = CaskLoader.load(old_cask.installed_caskfile) if old_cask.installed_caskfile.exist?
+          end
+
+          old_cask_installer = Installer.new(old_cask, binaries: binaries?, verbose: verbose?, force: force?, upgrade: true)
+
+          new_cask = CaskLoader.load(old_cask.to_s)
+
+          new_cask_installer =
+            Installer.new(new_cask, binaries:       binaries?,
+                                    verbose:        verbose?,
+                                    force:          force?,
+                                    skip_cask_deps: skip_cask_deps?,
+                                    require_sha:    require_sha?,
+                                    upgrade: true)
+
+          started_upgrade = false
+          new_artifacts_installed = false
+
+          begin
+            # Start new Cask's installation steps
+            new_cask_installer.check_conflicts
+
+            new_cask_installer.fetch
+
+            new_cask_installer.stage
+
+            # Move the old Cask's artifacts back to staging
+            old_cask_installer.start_upgrade
+            # And flag it so in case of error
+            started_upgrade = true
+
+            # Install the new Cask
+            new_cask_installer.install_artifacts
+            new_artifacts_installed = true
+
+            new_cask_installer.enable_accessibility_access
+
+            # If successful, wipe the old Cask from staging
+            old_cask_installer.finalize_upgrade
+          rescue CaskError => e
+            opoo e.message
```

Comment:
Instead outputting the error message here, re-raise the exception at the end. Otherwise it will exit with `0` when it fails.